examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -1058,17 +1048,52 @@ def load_model_hook(models, input_dir):
                 raise ValueError(f"unexpected save model: {model.__class__}")
 
         lora_state_dict, network_alphas = LoraLoaderMixin.lora_state_dict(input_dir)
-        LoraLoaderMixin.load_lora_into_unet(lora_state_dict, network_alphas=network_alphas, unet=unet_)
 
-        text_encoder_state_dict = {k: v for k, v in lora_state_dict.items() if "text_encoder." in k}
-        LoraLoaderMixin.load_lora_into_text_encoder(
-            text_encoder_state_dict, network_alphas=network_alphas, text_encoder=text_encoder_one_
-        )
+        unet_state_dict = {f'{k.replace("unet.", "")}': v for k, v in lora_state_dict.items() if k.startswith("unet.")}
+        unet_state_dict = convert_unet_state_dict_to_peft(unet_state_dict)
+        incompatible_keys = set_peft_model_state_dict(unet_, unet_state_dict, adapter_name="default")
+        if incompatible_keys is not None:
+            # check only for unexpected keys
+            unexpected_keys = getattr(incompatible_keys, "unexpected_keys", None)
+            if unexpected_keys:
+                logger.warning(
+                    f"Loading adapter weights from state_dict led to unexpected keys not found in the model: "
+                    f" {unexpected_keys}. "
+                )
 
-        text_encoder_2_state_dict = {k: v for k, v in lora_state_dict.items() if "text_encoder_2." in k}
-        LoraLoaderMixin.load_lora_into_text_encoder(
-            text_encoder_2_state_dict, network_alphas=network_alphas, text_encoder=text_encoder_two_
-        )
+        if args.train_text_encoder:
+            # Do we need to call `scale_lora_layers()` here?
+            text_encoder_state_dict = {
+                f'{k.replace("text_encoder.", "")}': v
+                for k, v in lora_state_dict.items()
+                if k.startswith("text_encoder.")
+            }
+            text_encoder_state_dict = convert_state_dict_to_peft(
+                convert_state_dict_to_diffusers(text_encoder_state_dict)
+            )
+            set_peft_model_state_dict(text_encoder_one_, text_encoder_state_dict, adapter_name="default")
+
+            text_encoder_2_state_dict = {
+                f'{k.replace("text_encoder_2.", "")}': v
+                for k, v in lora_state_dict.items()
+                if k.startswith("text_encoder_2.")
+            }
+            text_encoder_2_state_dict = convert_state_dict_to_peft(
+                convert_state_dict_to_diffusers(text_encoder_2_state_dict)
+            )
+            set_peft_model_state_dict(text_encoder_two_, text_encoder_2_state_dict, adapter_name="default")
+
+        # Make sure the trainable params are in float32. This is again needed since the base models
```

Comment:
Duplicate code? See line 1111

Reply:
It's not duplicate code, actually. 

What happens is that:

* You first load the models in the specified `weight_dtype`.
* You then explicitly cast the trainable params (LoRA params here more specifically) to be of `torch.float32`.
* Then when a checkpoint (LoRA checkpoint) is loaded to resume training from that point in time, they get casted to the dtype of the base model which is of `weight_dtype` at this point. 

This is why I believe this change is necessary.

Reply:
I see. Maybe this could be added as a comment to the script. The casting code itself is identical and could be factored into a function but it's not a big deal (we just added `cast_mixed_precision_params` to PEFT for exactly this purpose but that would be only available in future PEFT versions).

Reply:
Okay great!

> The casting code itself is identical and could be factored into a function

I will do this in a follow-up PR as this is going to be shared by a couple of scripts already. 

> we just added cast_mixed_precision_params to PEFT for exactly this purpose but that would be only available in future PEFT versions

Happy to graduate to using that. Do remind me once in the coming weeks?